src/main/java/de/tum/in/www1/artemis/repository/LectureUnitRepository.java

```diff
@@ -30,7 +30,18 @@ public interface LectureUnitRepository extends JpaRepository<LectureUnit, Long>
                 LEFT JOIN FETCH exercise.competencies
             WHERE lu.id = :lectureUnitId
             """)
-    Optional<LectureUnit> findByIdWithCompetencies(@Param("lectureUnitId") Long lectureUnitId);
+    Optional<LectureUnit> findWithCompetenciesById(@Param("lectureUnitId") Long lectureUnitId);
+
+    @Query("""
+            SELECT lu
+            FROM LectureUnit lu
+                LEFT JOIN FETCH lu.competencies
+                LEFT JOIN FETCH lu.exercise exercise
+                LEFT JOIN FETCH exercise.competencies
+                LEFT JOIN FETCH lu.slides
+            WHERE lu.id = :lectureUnitId
+            """)
+    Optional<LectureUnit> findWithCompetenciesAndSlidesById(@Param("lectureUnitId") Long lectureUnitId);
```

Comment:
```suggestion
    Optional<LectureUnit> findWithCompetenciesAndSlidesById(@Param("lectureUnitId") long lectureUnitId);
```
Also can we use an entity graph here?